### PR TITLE
build: use vite/vitest

### PR DESCRIPTION
**Pull request by treardon17**
| Type       | Description                                                                | Icon | Changelog |
| ---------- | ---------------------------------------------------------------------------| :--: | :-------: |
| `build`     | The build dependencies or scripts (package.json, etc...) are updated      |  🎁  |           |
-------------------------------------------------------------------------------------------------------------

### 🛠️ Changes:

- remove babel, jest, prettier, etc.
- use vite/vitest for consistency between packages
- update lint rules

### 📚 Bookkeeping:

**Testing (if applicable)**:
- [x] Ran/wrote unit tests for this

**Checklist**
- [x] Assigned PR to myself
- [x] Added at least 1 person on the team as reviewer
- [ ] Release Notes: PRs types that have the :spiral_notepad: next to them also require release notes to be added to the `CHANGELOG.md`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,27 +45,22 @@
...
     "eslint-config-airbnb-base": "^15.0.0",
-    "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-import": "^2.28.1",
-    "eslint-plugin-prettier": "^5.0.0",
     "eslint-plugin-unicorn": "^48.0.1",
-    "jest": "^29.6.4",
-    "jest-environment-jsdom": "^29.6.4",
+    "happy-dom": "^12.6.0",
     "lodash": "4.17.21",
     "prettier": "^3.0.3",
```
Comment (bitencode) [inline package.json:58]: looks like we don't need prettier to be installed any more, so should remove the dependency 😄

**Review by bitencode — APPROVED** [review]
👍 Looks good - I don't think we need the prettier dependency, other than that looks great.

File: package.json (hunk 1 above)
Comment (treardon17) [reply package.json:58]: good catch! missed that one 👍

**Review by bitencode — APPROVED** [review]
👍 Thanks! Looks good

**Pull request merged**